modules/apps/social/social-bookmarks-taglib/src/main/java/com/liferay/social/bookmarks/taglib/internal/util/SocialBookmarksRegistryImpl.java

```diff
@@ -77,19 +76,17 @@ public List<SocialBookmark> getSocialBookmarks() {
 
 	@Override
 	public List<String> getSocialBookmarksTypes() {
-		Set<String> socialBookmarksTypes = new LinkedHashSet<>();
-
-		for (String type : _serviceTrackerList) {
-			socialBookmarksTypes.add(type);
-		}
+		List<String> socialBookmarksTypes = _serviceTrackerList.toList();
```

Comment:
@dantewang @W2jx I do not think we need a _serviceTrackerList here, please remove it in a separate pull. Because of property "_SOCIAL_BOOKMARK_TYPES" had been removed, all the logic related to it can be removed, you can have a service tracker customizer for the service tracker map to gather the information you need here